src/services/item/repository.ts

```diff
@@ -103,7 +103,7 @@ export class ItemRepository {
     } = args;
 
     if (parent && !isItemType(parent, ItemType.FOLDER)) {
-      throw new ItemNotFolder(parent);
+      throw new ItemNotFolder(parent.id);
```

Comment:
I think it is inconsistent with your other throws (B.e MemberNotFound)
```suggestion
      throw new ItemNotFolder({ id: parent.id });
```

Reply:
Yes you are right, my issue is that with the object signature, it is possible to pass the full object and it will not complain. This could potentially lead to the same issue that created the need for this PR. 

I can use the object signature in all errors but then I need to explicitly strip out any additional properties that could be passed alongside. 

@Defferrard Do you see a better way of doing it ?

Reply:
From https://stackoverflow.com/questions/49580725/is-it-possible-to-restrict-typescript-object-to-contain-only-properties-defined someone suggest to create theses types :
```typescript
type Impossible<K extends keyof any> = {
  [P in K]: never;
};
type NoExtraProperties<T, U extends T = T> = U & Impossible<Exclude<keyof U, keyof T>>;
```

So we would to use it that way : 
```typescript
function a(s: NoExtraProperties<{a:string}>){
  return s
}
a({a:""}) // OK
a({}) // NOK: Property 'a' is missing in type '{}' but required in type '{ a: string; }'
a({a:"", b:""}) // NOK: Object literal may only specify known properties, and 'b' does not exist in type 
```

Fun fact, an issue is opened since 2016 https://github.com/microsoft/TypeScript/issues/12936, but there is still no native solution.